### PR TITLE
Update Go support to 1.6

### PR DESCRIPTION
**Pull request by cdonati**
This removes all of the Go source files from the repository. It downloads (and caches) the App Engine Go SDK and uses the binaries from that to build Go applications.

### CODE REVIEW

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/2511/

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/2541/

File: AppController/lib/helperfunctions.rb [hunk 1]
```diff
@@ -458,9 +458,16 @@ def self.setup_app(app_name, untar=true)
...
+    return unless untar
+
+    self.shell("tar --file #{tar_path} --force-local --no-same-owner " +
+      "-C #{tar_dir} -zx")
+
+    # Separate extra dependencies for Go applications.
+    begin
+      FileUtils.mv("#{tar_dir}/gopath", "#{meta_dir}/gopath")
+    rescue Errno::ENOENT
+      # Only Go applications with extra dependencies have this directory.
```
Comment (menivaitsi) [inline AppController/lib/helperfunctions.rb]: Can we have a debug log here instead of an empty rescue section?

**Comment by menivaitsi** [comment]
Depends on https://github.com/AppScale/appscale-tools/pull/570.

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/2611/

**Pull request merged**